resources/netlog_defs.ts

```diff
@@ -592,6 +602,15 @@ const latestLogDefinitions = {
     },
     canAnonymize: true,
     firstOptionalField: undefined,
+    analysisOptions: {
+      include: 'filter',
+      filters: { // DoT on player with valid effectId
+        id: '1.{7}',
+        which: 'DoT',
+        effectId: '[1-9A-F]+', // non-zero non-empty value
```

Comment:
I believe that this regex will fail with an id of e.g. `10D`? Should instead be `[1-9A-F][0-9A-F]*`?

Reply:
> I believe that this regex will fail with an id of e.g. `10D`? Should instead be `[1-9A-F][0-9A-F]*`?

I was thinking it would match the '1' (and since we don't capture, we don't care if it was only a partial match), but I forgot that `splitter` uses `buildRegex()` rather than just matching each field, so the field boundaries are excluding the match.

I think yours works assuming there is no padding on the field (e.g. `D` is always `D` and not `0D`).  If there is padding, we can do `[0-9A-F]{2,}` - or if we're not sure, then `[0-9A-F]{2,}|[1-9A-F]` is ~fugly af~ probably safer.

Reply:
`[0-9A-F]{2,}` still matches `|00|` in the assumption that there's padding.

If we need to account for potential padding in the future, it becomes something like `0*?[1-9A-F][0-9A-F]*`, test cases:

![image](https://github.com/OverlayPlugin/cactbot/assets/6119598/0f7be449-2d88-4770-835f-7fa37b89f90b)